### ISSUE TITLE
Need help with MLP tuning

### ISSUE THREAD

**Issue by dping1** [dmlc/mxnet#6546]
I am trying to train a simple MLP model for sentiment analysis using the IMDB dataset in MXNet.  I tried to duplicate the same MLP design and used the same hyper-parameters as the keras example in the link below.  However, my MXNet version does not seem to learn.  The training accuracy was always stuck around 50%, whereas the keras sample (using Tensorflow backend) produced an accuracy of 89%.

http://machinelearningmastery.com/predict-sentiment-movie-reviews-using-deep-learning/

My network has the following configurations and layers:
Data->Embedding->Flatten->FullConnected(250 hidden neurons)->Activation(ReLU)->FullConnected(1 hidden neuron)->LogisticRegressionOutput

Optimizer: adam
learning rate: 0.1
number of epochs: 10

Appreciate any help on this

**Comment by marugari** [comment]
Des `SoftmaxOutput` work well?
Maybe you can change your Network: 
FullConnected(2 hidden unit)->SoftmaxOutput

https://github.com/dmlc/mxnet/issues/3355

**Comment by dping1** [comment]
@marugari,  thanks for your feedback.  Just tried SoftmaxOutput and it worked.   Any idea why LogisticRegressionOutput would not work in this case?

**Comment by dping1** [comment]
@marugari Thanks again!  I am closing this issue.

**Comment by ryangarner** [comment]
The solution is to create a custom metric with a defined threshold. In this example, I set the threshold to 0.5.
```
import mxnet as mx
import numpy as np
import logging
import sys
logging.basicConfig(stream = sys.stdout, level = logging.DEBUG)

data = np.random.random((1000, 100))
label = np.random.randint(2, size = (1000, 1))

train_iter = mx.io.NDArrayIter(data = data,
                               label = label,
                               data_name = 'data',
                               label_name = 'label',
                               batch_size = 32,
                               shuffle = True)

data = mx.symbol.Variable('data')
label = mx.symbol.Variable('label')
fc1 = mx.symbol.FullyConnected(data = data, num_hidden = 32)
act = mx.symbol.Activation(data = fc1, act_type = 'relu')
fc2 = mx.symbol.FullyConnected(data = act, num_hidden = 1)
logits = mx.symbol.LogisticRegressionOutput(data = fc2, label = label)

# Custom metric with threshold set to 0.5
metric = mx.metric.CustomMetric(feval = lambda labels, pred: ((pred > 0.5) == labels).mean(),
                                name = "acc") 

model = mx.mod.Module(symbol = logits,
                      data_names = ['data'],
                      label_names = ['label'],
                      context = mx.gpu(0))

model.fit(train_data = train_iter,
          optimizer = 'rmsprop',
          eval_metric = metric,
          num_epoch = 10)

INFO:root:Epoch[0] Train-acc=0.498047
INFO:root:Epoch[0] Time cost=0.049
INFO:root:Epoch[1] Train-acc=0.517578
INFO:root:Epoch[1] Time cost=0.040
INFO:root:Epoch[2] Train-acc=0.535156
INFO:root:Epoch[2] Time cost=0.035
INFO:root:Epoch[3] Train-acc=0.556641
INFO:root:Epoch[3] Time cost=0.039
INFO:root:Epoch[4] Train-acc=0.590820
INFO:root:Epoch[4] Time cost=0.039
INFO:root:Epoch[5] Train-acc=0.604492
INFO:root:Epoch[5] Time cost=0.035
INFO:root:Epoch[6] Train-acc=0.603516
INFO:root:Epoch[6] Time cost=0.032
INFO:root:Epoch[7] Train-acc=0.611328
INFO:root:Epoch[7] Time cost=0.029
INFO:root:Epoch[8] Train-acc=0.610352
INFO:root:Epoch[8] Time cost=0.026
INFO:root:Epoch[9] Train-acc=0.620117
INFO:root:Epoch[9] Time cost=0.026
```
